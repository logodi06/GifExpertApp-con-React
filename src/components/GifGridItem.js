import React from 'react';

//Short cut para importar los proptypes
//impt
import PropTypes from 'prop-types';

 const GifGridItem = ( {title, url} ) => {
    //console.log(id, title, url );
    return (
        <div className='card animate__animated animate__fadeIn '>
            <img src={url} alt={title}/>
            <p> {title} </p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
} 

export default GifGridItem;

//El error que da al ejecutar el test 
//Cannot read property 'contextTypes' of undefined, hay que cambiar la forma en exportar
//el componente y hay que colocar el export default.