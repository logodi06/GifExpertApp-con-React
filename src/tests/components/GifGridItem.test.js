
import { shallow } from "enzyme";
import GifGridItem from '../../components/GifGridItem'
// import '@testing-library/jest-dom';
// import { setupTests } from "../../setupTests";


describe('Pruebas en GifExpertApp', () => {
    const title= 'Un título';
    const url = 'https://localhost/algo.jpg';
    
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );
  
    test('Debe mostrar el componente GifExpertApp correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un párrafo con el title', () => {
        const p = wrapper.find('p');
       expect(p.text().trim()).toBe(title); 
    }); 

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log(img.props('src'));
        
        //console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe de tener animate_fadeIn', () => {
        const div =  wrapper.find('div');
        const nameClass = (div.props().className);
        //console.log(nameClass);
        expect(nameClass.includes('animate__fadeIn')).toBe(true);
    });
    
    
    
    
})
