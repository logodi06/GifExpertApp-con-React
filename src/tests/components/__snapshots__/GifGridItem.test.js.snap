// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar el componente GifExpertApp correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
